### ISSUE TITLE
Tasklist widget border goes below screen when displaying east asian alphabets.

### ISSUE THREAD

**Issue by yobleck** [qtile/qtile#2370]
# Issue description
Tasklist widget border goes below screen when displaying east asian alphabets.

Steps to reproduce. Have taskList widget on bar. Open web browser. Type in east asian characters or copy them from wikipedia.

Tested languages include:
    - Japanese hiragana and katakana
    - Chinese pinyin
    - Thai
    - Burmese

No issue with Cyrillic or Arabic.
Possibly an issue with all unicode characters of non standard height?

Tested with  1920x1080_144 and 2560x1440_60 monitors.

![qtile_tasklist_issue](https://user-images.githubusercontent.com/21085003/114276428-89543800-99f4-11eb-9419-0328b61a859b.png)

Could this cause a crash if the monitor is at max screen size?
Note: This is not an issue with the group widget which accepts characters with a properly displayed border.

# Qtile version
0.17.0 AUR

# Stack traces
not sure if relevant.

# Configuration
```
screens = [
    Screen(
        bottom=bar.Bar(
            [
                widget.CurrentLayout(foreground="#00aa00"),
                widget.TextBox("|", foreground="#00aa00"),
                widget.GroupBox(active="#00aa00", inactive="#004400", block_highlight_text_color="#00aa00", disable_drag=True),
                widget.TextBox("|", foreground="#00aa00"),
                widget.Prompt(),

                #Relevant widget
                widget.TaskList(foreground="#00aa00", border="#00aa00"),


                widget.Chord( #multi key binds but not holding all keys down at same time
                    chords_colors={
                        'launch': ("#ff0000", "#ffffff"),
                    },
                    name_transform=lambda name: name.upper(),
                ),
                widget.Notify(foreground="#00aa00", foreground_low="#004400", max_chars=50),
                widget.Systray(),

                widget.CheckUpdates(custom_command="pamac checkupdates -q; echo -n",
                                    execute="pamac-manager", update_interval="600",
                                    no_update_string=" U ", colour_no_updates="#00aa00", colour_have_updates="#aa0000",
                                    ),
                widget.Clock(format='%Y-%m-%dT%H:%M', fontsize=18, foreground="#00aa00"), 
            ],
            24,
            #opacity=0,
            background=["#000000","#000000","#000000","#003300"], #bar background
        ),
    ),
```

**Comment by elParaguayo** [comment]
I think you're right. The issue seems to be that, because the characters are bigger, the text layout automatically increases its height to fit the text (currently, the code doesn't let you fix the height). The result is that the border around the layout is cut out.

I'll have a look and see if there's an easy fix.

**Comment by elParaguayo** [comment]
Nope. I was wrong. The layout's height isn't changing (stays constant on my machine when showing the full box and also when it's cut off).

Feels like this is more of an issue in how the TextLayout works. More digging...

**Comment by yobleck** [comment]
Temporary fix:
In the the user's config the tasklist properties `padding_y=1` and `margin_y=2` or 3 kind of fixes the issue on a 1080p monitor with
all other properties at default values, but it makes the icon overlap the border which looks weird.

I've also been messing around with pango markup. Changing the font size and rise values doesn't help and setting the font to a Japanese font doesn't work either.
